test/h3core.spec.js

```diff
@@ -1500,34 +1462,148 @@ test('edgeLength', assert => {
 
 test('edgeLength - bad units', assert => {
     const res = 9;
-    assert.throws(() => h3core.edgeLength(res), /Unknown/, 'throws on missing unit');
-    assert.throws(() => h3core.edgeLength(res, 'foo'), /Unknown/, 'throws on unknown unit');
-    assert.throws(() => h3core.edgeLength(res, 42), /Unknown/, 'throws on unknown unit');
-    assert.throws(
-        () => h3core.edgeLength(res, h3core.UNITS.km2),
-        /Unknown/,
-        'throws on invalid unit'
-    );
+    assert.throws(() => h3.edgeLength(res), /Unknown/, 'throws on missing unit');
+    assert.throws(() => h3.edgeLength(res, 'foo'), /Unknown/, 'throws on unknown unit');
+    assert.throws(() => h3.edgeLength(res, 42), /Unknown/, 'throws on unknown unit');
+    assert.throws(() => h3.edgeLength(res, h3.UNITS.km2), /Unknown/, 'throws on invalid unit');
 
     assert.end();
 });
 
 test('edgeLength - bad resolution', assert => {
-    assert.throws(
-        () => h3core.edgeLength(42, h3core.UNITS.m),
-        /Invalid/,
-        'throws on invalid resolution'
+    assert.throws(() => h3.edgeLength(42, h3.UNITS.m), /Invalid/, 'throws on invalid resolution');
+
+    assert.throws(() => h3.edgeLength(), /Invalid/, 'throws on invalid resolution');
+
+    assert.end();
+});
+
+test('cellArea', assert => {
+    // These values are slightly different from the comparable h3-java test,
+    // but match within some reasonable threshold
+    const expectedAreas = [
+        2562182.1629554993,
+        447684.20172018476,
+        65961.62242711011,
+        9228.87291900244,
+        1318.694490797148,
+        187.95935122814547,
+        26.87164354763051,
+        3.8408488470593176,
+        0.5486939641334774,
+        0.07838600808653977,
+        0.01119834222000042,
+        0.0015997771692175615,
+        0.00022853909314018362,
+        0.00003264850232866918,
+        0.000004664070325946983,
+        6.662957600331753e-7
+    ];
+    for (let res = 0; res < 16; res++) {
+        const h3Index = h3.geoToH3(0, 0, res);
+        const cellAreaKm2 = h3.cellArea(h3Index, h3.UNITS.km2);
+        assert.ok(
+            almostEqual(cellAreaKm2, expectedAreas[res]),
+            `Area matches expected value at res ${res}`
+        );
+        assert.ok(
+            // res 0 has high distortion of average area due to high pentagon proportion
+            res === 0 ||
+                // This seems to be the lowest factor that works for other resolutions
+                almostEqual(cellAreaKm2, h3.hexArea(res, h3.UNITS.km2), cellAreaKm2 * 0.4),
+            `Area is close to average area at res ${res}`
+        );
+        // Just property tests for the other units
+        assert.ok(h3.cellArea(h3Index, h3.UNITS.m2) > cellAreaKm2, 'm2 > Km2');
+        assert.ok(cellAreaKm2 > h3.cellArea(h3Index, h3.UNITS.rads2), 'Km2 > rads2');
```

Comment:
Not blocking, but why not take the expected areas from the average area function for the given resolution and then assert that the ratio of the two areas is below a maximum distortion factor?

Reply:
I do this for `Km`, but I assume you're asking why not do it for `m` too? I can add. I was copying `h3-java` here, and was lazy :).

No averages available for rads, though.